File: ios/runanywhere-sdks-latest/sdk/runanywhere-react-native/packages/core/src/Foundation/Constants/SDKConstants.ts
/**
 * SDK-wide constants (metadata only)
 *
 * Centralized constants to ensure consistency across the SDK.
 * Matches pattern: sdk/runanywhere-swift/Sources/RunAnywhere/Foundation/Constants/SDKConstants.swift
 */

import { Platform } from 'react-native';

/**
 * SDK Constants
 *
 * All SDK-wide constants should be defined here to avoid hardcoded values
 * scattered throughout the codebase.
 */
export const SDKConstants = {
  /**
   * SDK version - must match the VERSION file in the repository root
   * Update this when bumping the SDK version
   */
  version: '0.2.0',

  /**
   * SDK name
   */
  name: 'RunAnywhere SDK',

  /**
   * User agent string
   */
  get userAgent(): string {
    return `${this.name}/${this.version} (React Native)`;
  },

  /**
   * Platform identifier (ios/android)
   */
  get platform(): string {
    return Platform.OS === 'ios' ? 'ios' : 'android';
  },

  /**
   * Minimum log level in production
   */
  productionLogLevel: 'error',
} as const;

File: ios/runanywhere-sdks-latest/sdk/runanywhere-swift/Sources/RunAnywhere/Foundation/Constants/SDKConstants.swift
import Foundation

/// SDK-wide constants (metadata only)
/// Capability-specific constants are in their respective capabilities:
/// - LLMConstants (LLM capability)
/// - StorageConstants (FileManagement capability)
/// - DownloadConstants (Download capability)
/// - LifecycleConstants (Lifecycle capability)
/// - RegistryConstants (Registry capability)
public enum SDKConstants {
    /// SDK version - must match the VERSION file in the repository root
    /// Update this when bumping the SDK version
    public static let version = "0.16.0"

    /// SDK name
    public static let name = "RunAnywhere SDK"

    /// User agent string
    public static let userAgent = "\(name)/\(version) (Swift)"

    /// Platform identifier
    #if os(iOS)
    public static let platform = "ios"
    #elseif os(macOS)
    public static let platform = "macos"
    #elseif os(tvOS)
    public static let platform = "tvos"
    #elseif os(watchOS)
    public static let platform = "watchos"
    #else
    public static let platform = "unknown"
    #endif

    /// Minimum log level in production
    public static let productionLogLevel = "error"
}
